src/preact.d.ts

```diff
@@ -52,7 +52,7 @@ declare namespace preact {
 
 		linkState:(name:string) => (event: Event) => void;
 
-		setState(state:StateType, opts?:any):void;
+		setState(state:Partial<StateType>, opts?:any):void;
```

Comment:
This is cool. I've been making most of my State prop declarations optional, which is exactly what that generic type does https://github.com/Microsoft/TypeScript/pull/12276/files#diff-a6b488d9bd802977827b535a3011c1f3R1350

My only issue with using this is that it requires `typescript@next` if I'm not mistaken, and this could break some folks projects if released. I'm personally using `typescript@next`, so I'm good :)